subprojects/plugins/src/main/java/org/gradle/api/plugins/ApplicationPlugin.java

```diff
@@ -156,7 +156,7 @@ private void addRunTask(Project project, JvmSoftwareComponentInternal component,
             run.setClasspath(runtimeClasspath);
             run.getMainModule().set(pluginExtension.getMainModule());
             run.getMainClass().set(pluginExtension.getMainClass());
-            run.getConventionMapping().map("jvmArgs", pluginExtension::getApplicationDefaultJvmArgs);
+            run.getJvmArguments().convention(pluginExtension.getApplicationDefaultJvmArgs());
```

Comment:
⚠️ This is changing behavior. In the previous version, the final `Iterable` value of `pluginExtension.getApplicationDefaultJvmArgs()` would be captured in the convention while in the new version, the initial `Iterable` value would be captured in the convention which could be wrong if it was later [changed](https://github.com/gradle/gradle/blob/99f85445b6c9cf89d188a5f74dbfb2e409d0ebd8/subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultApplicationPluginConvention.java#L77-L76).
```suggestion
            run.getJvmArguments().convention(pluginExtension::getApplicationDefaultJvmArgs);
```